wandb/__init__.py

```diff
@@ -717,6 +717,16 @@ def init(job_type=None, dir=None, config=None, project=None, entity=None, reinit
         os.environ[env.TAGS] = ",".join(tags)
     if id:
         os.environ[env.RUN_ID] = id
+        if name is None:
```

Comment:
I think this logic overrides setting name by environment variable.

Reply:
Ah yeah I think it does. Will fix.

Reply:
Fixed:

```shell
python -c 'import wandb; wandb.init(id="asdfasde")'
```

https://app.wandb.ai/adrianbg-test/wandb-client-standalone_tests/runs/asdfasde

```shell
WANDB_NAME=ieslkd python -c 'import wandb; wandb.init(id="asdfasdf")'
```

https://app.wandb.ai/adrianbg-test/wandb-client-standalone_tests/runs/asdfasdf